diffrax/solver/ansr.py

```diff
@@ -0,0 +1,220 @@
+        assert self.b.ndim == 1
+        assert self.c.shape[0] == len(self.a)
+        assert all(i + 1 == a_i.shape[0] for i, a_i in enumerate(self.a))
+        assert self.c.shape[0] + 1 == self.b.shape[0]
+        assert self.cw.shape[0] == self.b.shape[0]
+        assert self.ch.shape[0] == self.b.shape[0]
+        for i, (a_i, c_i) in enumerate(zip(self.a, self.c)):
+            assert np.allclose(sum(a_i), c_i)
+        assert np.allclose(sum(self.b), 1.0)
+
+        # TODO: add checks for whether the method is FSAL
+
+
+class ANSR(AbstractItoSolver):
+    r"""Additive-Noise Stochastic Runge-Kutta method.
+
+    The second term in the MultiTerm must be a Control term with
+    control=VirtualBrownianTree(compute_stla=True), since this method
+    makes use of space-time Levy area.
+
+    Given the SDE
+    $dX_t = f(t, X_t) dt + σ dW_t$
+
+    We construct the SRK as follows:
+
+    $y_1 = y_0 + h \Big(\sum_{j=1}^s b_j k_j \Big) + σ \, (c^W_{s+1} ΔW + c^H_{s+1} ΔH)$
+
+    $k_j = f(t_0 + c_j h , z_j)$
+
+    $z_j = y_0 + h \Big(\sum_{i=1}^{j-1} a_{j,i} k_i \Big) + σ \, (c^W_j ΔW + c^H_j ΔH)$
+
+    where $ΔW := W_{t_0, t_1}$ is the increment of the Brownian motion and
+    $ΔH := H_{t_0, t_1}$ is its corresponding space-time Levy Area.
+
+    The values $( a_{i,j} ) , b_j, c_j, c^W_j, c^H_j$ are provided
+    in the StochasticButcherTableau.
+    """
+
+    term_structure = MultiTerm[Tuple[ODETerm, ControlTerm]]
+    interpolation_cls = LocalLinearInterpolation
+    tableau: StochasticButcherTableau
+
+    def order(self, terms):
+        return 1  # should be modified depending on tableau
+
+    def strong_order(self, terms):
+        return 0.5  # should be modified depending on tableau
+
+    def init(
+        self,
+        terms: MultiTerm[Tuple[ODETerm, ControlTerm]],
+        t0: Scalar,
+        t1: Scalar,
+        y0: PyTree,
+        args: PyTree,
+    ) -> _SolverState:
+        return None
+
+    def embed_a_lower(self, dtype):
+        """
+        Takes the lower-triangular (a_j_i) and embeds it in an n*n array.
+        """
+        num_stages = len(self.tableau.b)
+        a = self.tableau.a
+        tab_a = np.zeros((num_stages, num_stages))
+        for i, a_i in enumerate(a):
+            tab_a[i + 1, : i + 1] = a_i
+        return jnp.asarray(tab_a, dtype=dtype)
+
+    def step(
+        self,
+        terms: MultiTerm[Tuple[ODETerm, ControlTerm]],
+        t0: Scalar,
+        t1: Scalar,
+        y0: PyTree,
+        args: PyTree,
+        solver_state: _SolverState,
+        made_jump: Bool,
+    ) -> Tuple[PyTree, _ErrorEstimate, DenseInfo, _SolverState, RESULTS]:
+        del solver_state, made_jump
+
+        h = t1 - t0
+        drift, diffusion = terms.terms
+        check_diffusion_has_stla(diffusion)
+
+        # compute the Brownian increment and space-time Levy area
+        levy: LevyVal = diffusion.levy_contr(t0, t1)
+        w = levy.W
+        hh = levy.H
+        sigma = diffusion.vf(t0, y0, args)
```

Comment:
If you decide to stick with specifically additive noise, then that's actually an invariant we can check:
```python
sigma, t_sigma = eqx.filter_jvp(lambda y: diffusion.vf(t0, y, args), y0)
if t_sigma is not None:
    raise ValueError
```
The above checks that `t_sigma` is returning a symbolic zero tangent when differentiated, i.e. it doesn't depend on the `y0` input.

Reply:
I did add that check above.